### PR TITLE
Proofpoint Protection Server v2

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/25330

## Minimum version of Demisto
- [x] 5.0.0

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2_description.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+### Authentication
+An administrator must have a role that includes access to a specific REST API. 
+
+Proofpoint on Demand (PoD) administrators must file a support ticket to Proofpoint support to obtain a role with access to an API.
+
+On premise administrators: Edit the **filter.cfg** file and set the following key to true: com.proofpoint.admin.apigui.enable=t
```
Comment (yaakovi) [inline Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2_description.md:6]: `=t`? or are you missing the rest of the word?

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+            return_results(resubmit_message(client, demisto.args()))
+        elif command == 'proofpoint-pps-forward-message':
+            return_results(forward_message(client, demisto.args()))
+        elif command == 'proofpoint-pps-move-message':
+            return_results(move_message(client, demisto.args()))
+        elif command == 'proofpoint-pps-delete-message':
+            return_results(delete_message(client, demisto.args()))
+        elif command == 'proofpoint-pps-download-message':
+            return_results(download_message(client, demisto.args()))
+
```
Comment (yaakovi) [inline Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.py:310]: add `else` with `NotImplementedError` (new convention)

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.py [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+                'guid': guid,
+            },
+            resp_type='response',
+            ok_codes=(200, 404),
+        )
+
+
+def test_module(client: Client) -> str:
+    client.health_check()  # test pss managed module
+    client.list_quarantined_messages_request(subject='Test')  # test Quarantine managed module
```
Comment (yaakovi) [inline Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.py:136]: nice

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.py [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+
+def main() -> None:
+    try:
+        command = demisto.command()
+        params = demisto.params()
+        client = Client(
+            base_url=urljoin(params['url'], '/rest/v1'),
+            auth=(params['credentials']['identifier'], params['credentials']['password']),
+            verify=not params.get('insecure', False),
+            proxy=params.get('proxy', False),
```
Comment (yaakovi) [inline Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.py:290]: I think you still need to call `handle_proxy`

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/README.md [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+#### Base Command
+
+`proofpoint-pps-smart-search`
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| action | Final disposition action message. Possible values are: "accept", "continue", "discard", "redirect", "reject", "retry". | Optional | 
+| start_time | Time from when the search should begin. Can be either free text (<number> <time unit>, e.g., 12 hours, 7 days) or ISO-8601 (YYYY-MM-DDThh:mm:ssZ, e.g., 2020-01-25T10:00:00Z). Default is "24 hours". | Optional | 
+| end_time | Time of when the search should end. Can be either free text (<number> <time unit>, e.g., 12 hours, 7 days) or ISO-8601 (YYYY-MM-DDThh:mm:ssZ, e.g., 2020-01-25T10:00:00Z). | Optional |
```
Comment (yaakovi) [inline Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/README.md:52]: the `<>` chars are failing the build

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.yml [hunk 6]
```diff
@@ -0,0 +1,548 @@
...
+    - default: false
+      description: The new subject with which to overwrite the original subject for the message.
+      isArray: false
+      name: subject
+      required: false
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'false'
+      description: 'Whether to append the original subject to the string specified in the subject argument. Possible values are: "true" and "false". Default is "false".'
```
Comment (yaakovi) [inline Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.yml:437]: no need to specify defaults, the server adds them automatically

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.yml [hunk 7]
```diff
@@ -0,0 +1,548 @@
...
+    - default: false
+      description: |-
+        The ID of the header message. This corresponds to the Message ID field in the UI.
+      isArray: false
+      name: message_id
+      required: false
+      secret: false
+    - default: false
+      defaultValue: '100'
+      description: The maximum number of email messages to return. The maximum value is set by the com.proofpoint.pss.query.default-count value in the filter.cfg file. Default is "100".
```
Comment (yaakovi) [inline Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.yml:118]: same here

**Review by yaakovi — COMMENTED** [review]
how was this tested?
no test playbook (you have `No test` in the yml)

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2_description.md (hunk 1 above)
Comment (Itay4) [reply Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2_description.md:6]: `=t`

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.py (hunk 2 above)
Comment (Itay4) [reply Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.py:310]: isnt just for mirroring ?

File: Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.yml (hunk 6 above)
Comment (Itay4) [reply Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.yml:437]: @ShirleyDenkberg FYI

**Comment by Itay4** [comment]
@yaakovi please re-review

**Review by yaakovi — APPROVED** [review]

**Pull request merged**